File: single-number.go
// 1 2 3 4 5 4 2 5 1 | 3
// 2 (1 2 4 5)

// 5 2 5 3 2
// 17 -- 3

// 0 1 0 1 
// 0 1 0 1

// xor them all!!!

func singleNumber(nums []int) int {
    ans := 0
    for _, x := range nums {
        ans ^= x
    }
    return ans
}

File: lc/py/single-number.py
# 1 4 2 1 2
# xor maybe? 
# 000
# 001
# 001
# 100
# 101
# 010
# 111
# 001
# 110
# 100
# just xore them all!

# another approach
# 1 2 1
# ans 10
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0
        for num in nums:
            ans ^= num
        return ans